### PR TITLE
creating single category component

### PR DESCRIPTION
**Pull request by sardorboboshov**
https://user-images.githubusercontent.com/89270394/163873881-1015aeba-e96e-476c-9f64-bd302e026286.mp4

I added updating functionality to category item. But when I change category title, it doesnt show updated version of category automatically, but when I refresh the page, it shows edited version of category. How can I solve that problem?

### CODE REVIEW

File: client/src/app/main/category/single-category/single-category.component.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  <form [formGroup]="categoryForm" *ngIf="!viewMode">
+    <input
+      class="category__title"
+      [value]="category.title"
+      formControlName="title"
+      [class.edit]="viewMode"
+    />
+  </form>
+  <div class="title" *ngIf="viewMode">
+    <span>{{ category.title | titlecase }}</span>
```
Comment (vokanred) [inline client/src/app/main/category/single-category/single-category.component.html:14]: You display category title but should display form control value

File: client/src/app/main/category/single-category/single-category.component.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<mat-card
+  class="container__wrapper"
+  [ngClass]="category.type === 'income' ? 'income' : 'expense'"
+>
+  <form [formGroup]="categoryForm" *ngIf="!viewMode">
+    <input
+      class="category__title"
+      [value]="category.title"
```
Comment (vokanred) [inline client/src/app/main/category/single-category/single-category.component.html:8]: You shouldn't bind value when use formControlName

File: client/src/app/main/category/single-category/single-category.component.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  selector: 'app-single-category',
+  templateUrl: './single-category.component.html',
+  styleUrls: ['./single-category.component.scss']
+})
+export class SingleCategoryComponent {
+  @Input() category!: ICategory;
+  viewMode: boolean = true;
+
+  categoryForm = new FormGroup({
+    title: new FormControl({ value: '', disabled: !this.viewMode }, [
```
Comment (vokanred) [inline client/src/app/main/category/single-category/single-category.component.ts:20]: You should update this value on category change

**Review by vokanred — CHANGES_REQUESTED** [review]
FormControl is a source of truth

**Comment by vokanred** [comment]
Screen Shot 2022-04-19 at 8 29 57 AM
Screen Shot 2022-04-19 at 8 30 12 AM

**Comment by vokanred** [comment]
Another solution is using template driven forms

**Comment by vokanred** [comment]
And one more solution is to update category object using response from API.

**Comment by sardorboboshov** [comment]
> And one more solution is to update category object using response from API.

It is working perfectly, thank you very much!

**Review by vokanred — APPROVED** [review]
Approved

**Pull request merged**